### PR TITLE
update the signature of from_single_file

### PR DESCRIPTION
**Pull request by yiyixuxu**
updating the signature of `from_single_file`! 
let me know if I missed anything 

motivated from issues like this https://github.com/huggingface/diffusers/discussions/7163

### CODE REVIEW

File: src/diffusers/loaders/single_file.py [hunk 1]
```diff
@@ -181,6 +181,31 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
...
+            original_config_file (`str`, *optional*):
+                The path to the original config file that was used to train the model. If not provided, the config file
+                will be inferred from the checkpoint file.
+            model_type (`str`, *optional*):
+                The type of model to load. If not provided, the model type will be inferred from the checkpoint file.
+            image_size (`int`, *optional*):
+                The size of the image output. It's used to configure the `simple_size` parameter of the UNet and VAE model.
+            load_safety_checker (`bool`, *optional*, defaults to `False`):
+                Whether to load the safety checker model or not. By default, the safety checker is not loaded unless user pass
+                a `safety_checker` component in the `kwargs`.
```
Comment (sayakpaul) [inline src/diffusers/loaders/single_file.py:193]: Should we maybe include a note that this doesn't apply to models like SDXL or even better something like "This doesn't apply to all models. Such as: SDXL." WDYT?

**Review by sayakpaul — APPROVED** [review]
Much needed. Ufff!

File: src/diffusers/loaders/single_file.py [hunk 2]
```diff
@@ -181,6 +181,31 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
             revision (`str`, *optional*, defaults to `"main"`):
                 The specific model version to use. It can be a branch name, a tag name, a commit id, or any identifier
                 allowed by Git.
+            original_config_file (`str`, *optional*):
+                The path to the original config file that was used to train the model. If not provided, the config file
+                will be inferred from the checkpoint file.
+            model_type (`str`, *optional*):
+                The type of model to load. If not provided, the model type will be inferred from the checkpoint file.
+            image_size (`int`, *optional*):
+                The size of the image output. It's used to configure the `simple_size` parameter of the UNet and VAE model.
```
Comment (stevhliu) [inline src/diffusers/loaders/single_file.py:190]:
```suggestion
                The size of the image output. It's used to configure the `sample_size` parameter of the UNet and VAE model.
```

File: src/diffusers/loaders/single_file.py (hunk 1 above)
Comment (stevhliu) [inline src/diffusers/loaders/single_file.py:193]:
```suggestion
                Whether to load the safety checker model or not. By default, the safety checker is not loaded unless a `safety_checker` component is passed to the `kwargs`.
```

File: src/diffusers/loaders/single_file.py [hunk 3]
```diff
@@ -181,6 +181,31 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
...
+            load_safety_checker (`bool`, *optional*, defaults to `False`):
+                Whether to load the safety checker model or not. By default, the safety checker is not loaded unless user pass
+                a `safety_checker` component in the `kwargs`.
+            num_in_channels (`int`, *optional*):
+                Specify the number of input channels for the UNet model. Read more about how to configure UNet model with this parameter
+                here https://huggingface.co/docs/diffusers/training/adapt_a_model#configure-unet2dconditionmodel-parameters
```
Comment (stevhliu) [inline src/diffusers/loaders/single_file.py:196]:
```suggestion
                [here](https://huggingface.co/docs/diffusers/training/adapt_a_model#configure-unet2dconditionmodel-parameters).
```

File: src/diffusers/loaders/single_file.py [hunk 4]
```diff
@@ -181,6 +181,31 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
...
+            scaling_factor (`float`, *optional*):
+                The scaling factor to use for the VAE model. If not provided, It will first try to infer from the config file.
+                If the scaling factor is not found in the config file, the defaut value 0.18215 is used
```
Comment (stevhliu) [inline src/diffusers/loaders/single_file.py:199]:
```suggestion
                The scaling factor to use for the VAE model. If not provided, it is inferred from the config file first.
                If the scaling factor is not found in the config file, the default value 0.18215 is used.
```

**Review by stevhliu — APPROVED** [review]
Thanks so much for the update!

**Review by DN6 — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**